server/komodo-rest/src/main/java/org/komodo/rest/relational/response/virtualization/RestVirtualizationStatus.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * See the COPYRIGHT.txt file distributed with this work for information
+ * regarding copyright ownership.  Some portions may be licensed
+ * to Red Hat, Inc. under one or more contributor license agreements.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA.
+ */
+package org.komodo.rest.relational.response.virtualization;
+
+import java.net.URI;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import org.komodo.rest.AbstractKEntity;
+import org.komodo.servicecatalog.BuildStatus;
+import org.komodo.spi.KException;
+import org.komodo.spi.repository.KomodoObject;
+import org.komodo.spi.repository.Repository.UnitOfWork;
+import org.komodo.utils.ArgCheck;
+
+public class RestVirtualizationStatus extends AbstractKEntity {
```

Comment:
AbstractKEntity? I thought you said if we are not storing in Komodo repo we do not need it?

Reply:
Nothing is being stored in the Komodo repo. Please see the hierarchy of those classes that inherit from KRestEntity.

Objects that do get stored in the repository tend to inherit from RestBasicEntity since this class included methods such as setId(), setDataPath() etc.

## Reason for Change
* getVirtualizations() returns a RestVirtualizationState (renamed from RestBasicState)
* With no serializer the standard json for the KRestEntity is returned in an http response, ie.
```json
{
  tuples: {
    buildName: 'blah'
    vdbName: 'blah'
    ....
  }
}
```

* In beetle-studio ts-script allows a very quick mechanism for turning json structures into javascript objects ('classes'). As such, a virtualization model has been added for tracking the getVirtualization() responses.
* With the extra level of tuples in the json response, it becomes harder to change the response into a virtualization model. Can be done but it is far easier with a serializer on the komodo side.
* The serializer removes the extra level and returns the json as:
```json
[
  buildName: 'blah'
  vdbName: 'blah'
  ....
],
[
....
]
```

* The response becomes more useful and easy to convert with javascript.

## Conclusion
1. A serializer was required due to a custom conversion;
2. The RestVirtualizationStatus is NOT being stored in the repository;

Reply:
I wish your default "properties" value encoding to JSON converted to an array instead of JSON object to avoid the custom serializers. I thought I added the values to properties, not tuples, and that could be the distinction between uses of when to use which one in the AbstractEntity classes.